src/npcmove.cpp

```diff
@@ -2661,17 +2679,23 @@ bool npc::saw_player_recently() const
              last_seen_player_turn > 0 );
 }
 
+bool npc::has_investigation_interest() const
+{
+    return investigation_target != no_goal_point && investigation_time > 0;
+}
+
 bool npc::has_destination() const
 {
     return goal != no_goal_point;
 }
 
-void npc::reach_destination()
+void npc::reach_destination(tripoint &new_target)
```

Comment:
The parameter is modified. I parameterized this object. I'll probably end up reverting this change since it became unnecessary/unused.